src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -3384,27 +3383,27 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 			});
 	}
 
-	private _runConfigureDefaultBuildTask(): void {
-		if (this.schemaVersion === JsonSchemaVersion.V2_0_0) {
-			this.tasks().then((tasks => {
+	private _runConfigureDefaultBuildTask(that: AbstractTaskService): void {
+		if (that.schemaVersion === JsonSchemaVersion.V2_0_0) {
```

Comment:
`runSingleTask` is a regular function, not a method though. We shouldn't need to use the `that` pattern here and this change introduces a bunch of debt imo. `this` on a method like this should only be undefined or not the expected value if you set the function like this without bind:

```ts
const f = this._something();
f(); // this = undefined
```

Or if you're in a different object scope, for example inside `function () { ... }` this will be the function.

We should identify why this goes missing, and probably add `.bind` instead of introducing `that` variables all over the place.

Reply:
It's not clear after skimming the PR why `this` isn't available, I'd recommend finding the first time it's undefined and stepping through to see why

Reply:
it's first undefined when that function is called, which is why it has to be passed in just like with `runSingleTask` mentioned in the image above.

Reply:
I just had another report of this bug and think we need this fix in the next release.i'll see what I can do to change the code around to avoid `this` and `that` passing

Reply:
but might be what we have to go with for now

Reply:
spent more time investigating and no way to avoid doing this AFAIK. maybe the bind thing you mentioned?

Reply:
interesting so `this` is defined for runSingleTask but it is undefined within the `if` where `configure` is called
![Screen Shot 2022-09-28 at 11 53 19 AM](https://user-images.githubusercontent.com/29464607/192865350-6af50edb-6ec0-442f-8cef-18eb8d9b07b9.png)

Reply:
tried `configure.bind(this);` above there outside of the `promise` and still didn't work

Reply:
`configure.bind(this).apply(this)` works